### PR TITLE
git-privacy: init at 2.0.0

### LINKED ISSUE #83692: Package request: git-privacy
**Project description**
Git-privacy redacts author and committer dates to keep your coding hours more private. You can choose the level of redaction: only remove minutes and seconds from your dates or even hide day or month. The original dates are encrypted and stored in the commit message in case you might need them.

**Metadata**

* homepage URL: https://pypi.org/project/gitprivacy/ 
* source URL: https://github.com/EMPRI-DEVOPS/git-privacy
* license: bsd2
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
Closes #83692

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/git-privacy/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    sha256 = "1djrhyxssv44cyakjx1cvy3agif1b1jzd0kgf4wh3dgsbwa4x3nn";
+  };
+
+  propagatedBuildInputs = [
+    git-filter-repo
+    GitPython
+    pynacl
+  ];
+
+  meta = with lib; {
```
Comment (mlvzk) [inline pkgs/tools/misc/git-privacy/default.nix:17]: lib is not defined in the first line

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+ { lib, buildPythonPackage, fetchPypi, setuptools_scm, click }:
```
Comment (jonringer) [inline pkgs/development/python-modules/git-filter-repo/default.nix:1]: NIT
```suggestion
 { lib, buildPythonPackage, fetchPypi, setuptools_scm, click }:
 
```

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+   src = fetchPypi {
+     inherit pname version;
+     sha256 = "0zgvbyvj3i1k9dkrzhw8r9jrylyb0454ysac8209lj46hcmzfdcv";
+   };
+
+   propagatedBuildInputs = [
+     click
+     setuptools_scm
+   ];
```
Comment (jonringer) [inline pkgs/development/python-modules/git-filter-repo/default.nix:14]: `setuptools_scm` is only needed at build time
```suggestion
   nativeBuildInputs = [
     setuptools_scm
   ];
   
   propagatedBuildInputs = [
     click
   ];
```

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+   meta = with lib; {
+     description = "Quickly rewrite git repository history";
+     license = licenses.mit;
+     homepage = "https://github.com/newren/git-filter-repo";
```
Comment (jonringer) [inline pkgs/development/python-modules/git-filter-repo/default.nix:19]: please add yourself as maintainer

File: pkgs/tools/misc/git-privacy/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonApplication, fetchPypi, pynacl, GitPython, git-filter-repo}:
```
Comment (jonringer) [inline pkgs/tools/misc/git-privacy/default.nix:1]:
```suggestion
{ lib, buildPythonApplication, fetchPypi, pynacl, GitPython, git-filter-repo}:

```

File: pkgs/tools/misc/git-privacy/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  propagatedBuildInputs = [
+    git-filter-repo
+    GitPython
+    pynacl
+  ];
+
+  meta = with lib; {
+    description = "Redact Git author and committer dates to keep committing behaviour more private.";
+    license = licenses.bsd2;
+    homepage = "https://github.com/EMPRI-DEVOPS/git-privacy";
```
Comment (jonringer) [inline pkgs/tools/misc/git-privacy/default.nix:20]: please add yourself as maintainer

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildPythonPackage, fetchPypi, setuptools_scm, click }:
```
Comment (jonringer) [inline pkgs/development/python-modules/git-filter-repo/default.nix:1]:
```suggestion
{ lib, buildPythonPackage, fetchPypi, isPy27, setuptools_scm, click }:
```

File: pkgs/development/python-modules/git-filter-repo/default.nix [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+buildPythonPackage rec {
+  pname = "git-filter-repo";
+  version = "2.28.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/git-filter-repo/default.nix:5]:
```suggestion
  version = "2.28.0";
  disabled = isPy27;
```

**Review by jonringer — COMMENTED** [review]
```
  NameError: name 'FileExistsError' is not defined
cannot build derivation '/nix/store/j8awllcjjpqhkvfwv52wv39vak3w3aqc-env.drv': 1 dependencies couldn't be built
[0 built (1 failed), 0.0 MiB DL]
error: build of '/nix/store/j8awllcjjpqhkvfwv52wv39vak3w3aqc-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/98773
1 package failed to build:
python27Packages.git-filter-repo

3 packages built:
git-privacy python37Packages.git-filter-repo python38Packages.git-filter-repo
```

**Comment by jonringer** [comment]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

I would use `git rebase -i`, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Comment by ajs124** [comment]
I could also just squash them when merging, if you approve of the diff.

**Comment by jonringer** [comment]
you have two package additions, they should be two separate commits. squash and merge will just make it into one

**Pull request closed without merging**